### DISCUSSION TITLE
Javascript code doesn't work

### DISCUSSION THREAD

**Discussion by matheuscostadesign** [requarks/wiki#4122] [Help / Questions] [answered]
Good afternoon people

I'm trying to use a simple javascript function to show an alert on the screen, but it's not working/rendering.

The script is this:

`<button onclick="myFunction()">Try it</button>`

`<script>
function myFunction() {
alert("Hello! I am an alert box!");
}
</script>`

What do I have to do to run any javascript function on the page?

**Comment by NGPixel** [comment, marked as answer]
You need to insert your javascript functions inside the Page Properties > Scripts tab. You will also need to disable HTML Sanitization under Admin > Rendering > HTML > Security in order for your onclick property to be kept.

**Reply by chenzhiqiang0518** [reply]
however, below script is not used, please why?
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>HTTP POST demo</title>
</head>
<body>
    <h1>HTTP POST demo</h1>
    <button id="sendRequestButton">send request</button>

    <script>
    document.getElementById("sendRequestButton").addEventListener("click", function() {
        const url = "http://127.0.0.1:7002/example";  // my demo api
        const data = {
            "key": "value"
        };

        fetch(url, {
            method: "POST",
            headers: {
                "Content-Type": "application/json"
            },
            body: JSON.stringify(data)
        })
        .then(response => response.json())
        .then(data => {
            alert("success: " + JSON.stringify(data));
        })
        .catch((error) => {
            console.error("error:", error);
        });
    });
    </script>
</body>
</html>

**Reply by NGPixel** [reply]
https://docs.requarks.io/en/troubleshooting#how-to-execute-javascript-code-on-page-load
